lib/utils/sortingUtils.js

```diff
@@ -0,0 +1,103 @@
+
+export function sortByType(array, type) {
+
+  // A custom sort function that sorts elements with $type equal to "type" to the front
+  return array.sort((a, b) => {
+    if (a.$type == type && b.$type != type) {
+      return -1;
+    }
+    if (a.$type != type && b.$type == type) {
+      return 1;
+    }
+    return 0;
+  });
+}
+
+export function sortByPosition(grid, array) {
+  return array.sort((a, b) => {
+    const aPosition = grid.find(a);
+    const bPosition = grid.find(b);
+    if (aPosition[1] !== bPosition[1]) {
+
+      // smallest column to the back to be popped first
+      return Math.sign(bPosition[1] - aPosition[1]);
+    } else {
+
+      // highest row to the back to be popped first
+      return Math.sign(bPosition[0] - aPosition[0]);
+    }
+  });
+}
+
+export function sortBySplittingMergingAndNone(array) {
```

Comment:
@till-stadtler It would be benefitial to explain what this function does, and name the argument in a semantic manner.

Reply:
This function is used for sorting already placed elements before revisiting them (either skipped or reversedSkipped). Elements that are neither splitting (boundary), or merging (coined "none") are sorted to the back to be popped first. Subsequently, splitting elements should be visited sooner than elements with boundary events. (I think this was the result of some trial and error, it should be revisited when thoroughly testing the tool)

Reply:
done